sdk/elixir/runtime/main.go

```diff
@@ -0,0 +1,177 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"path"
+
+	"github.com/iancoleman/strcase"
+)
+
+const (
+	ModSourceDirPath     = "/src"
+	sdkSrc               = "/sdk"
+	genDir               = "sdk"
+	schemaPath           = "/schema.json"
+	defaultElixirVersion = "1.16.1-erlang-26.2.2-debian-bookworm-20240130-slim"
+)
+
+func New(
+	// +optional
+	sdkSourceDir *Directory,
+) *ElixirSdk {
+	return &ElixirSdk{SDKSourceDir: sdkSourceDir, RequiredPaths: []string{}}
+}
+
+type ElixirSdk struct {
+	SDKSourceDir  *Directory
+	RequiredPaths []string
+}
+
+func (m *ElixirSdk) ModuleRuntime(
+	ctx context.Context,
+	modSource *ModuleSource,
+	introspectionJson string,
+) (*Container, error) {
+	ctr, err := m.CodegenBase(ctx, modSource, introspectionJson)
+	if err != nil {
+		return nil, err
+	}
+
+	modName, err := modSource.ModuleName(ctx)
+	if err != nil {
+		return nil, err
+	}
+	subPath, err := modSource.SourceSubpath(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	mod := normalizeModName(modName)
+	entrypoint := path.Join(ModSourceDirPath, subPath, mod)
+
+	return ctr.
+		WithEntrypoint([]string{
+			"mix", "cmd",
+			"--cd", entrypoint,
+			"mix do deps.get + dagger.invoke",
+		}), nil
+}
+
+func (m *ElixirSdk) Codegen(
+	ctx context.Context,
+	modSource *ModuleSource,
+	introspectionJson string,
+) (*GeneratedCode, error) {
+	ctr, err := m.CodegenBase(ctx, modSource, introspectionJson)
+	if err != nil {
+		return nil, fmt.Errorf("could not load module config: %v", err)
+	}
+
+	return dag.GeneratedCode(ctr.Directory(ModSourceDirPath)).
+		WithVCSGeneratedPaths([]string{genDir + "/**"}).
+		WithVCSIgnoredPaths([]string{"dagger"}), nil
+}
+
+func (m *ElixirSdk) CodegenBase(
+	ctx context.Context,
+	modSource *ModuleSource,
+	introspectionJson string,
+) (*Container, error) {
+	modName, err := modSource.ModuleName(ctx)
+	if err != nil {
+		return nil, err
+	}
+	subPath, err := modSource.SourceSubpath(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	mod := normalizeModName(modName)
+
+	codegenDepsCache, codegenBuildCache := mixProjectCaches(dag, "dagger-codegen")
+
+	ctr := m.Base("").
+		WithMountedDirectory(ModSourceDirPath, modSource.ContextDirectory()).
+		WithMountedDirectory(sdkSrc, m.SDKSourceDir).
```

Comment:
You don't seem to need the sdk in `/src`. Just put it in `genDir` (`<context directory> + <source subpath> + `sdk/`). You already seem to be doing it below but using `dagger` instead of `sdk`.

Reply:
Fixed in https://github.com/dagger/dagger/pull/6967/commits/1764f20086747660408ced5ef45b8b9db65b242e